examples/resolve.cpp

```diff
@@ -217,9 +193,13 @@ class Crc {
     JS::RootedObject proto(
         cx, JS_InitClass(cx,
                          global,   // the object in which to define the class
+                         nullptr,  // the prototype object, Crc.prototype
+                                   // (in our case, a plain JS object
+                                   // because calling "Crc.prototype.update"
+                                   // does not make sense)
```

Comment:
Not sure I understand this correctly! Does this mean that `Crc.prototype` doesn't have an `update` property but you can still call `update()` on an instance of `Crc`?

Reply:
Yes, I've tried adding `Crc.prototype.update()` into testProgram and it's properly rejected.

Reply:
@ptomato `Crc.prototype` does (still) have the `update` property defined on it, but it's just a plain object now so checking for the `JSClass` will reject the prototype object itself.

It's great to see the changes in bug 1808171 are resulting in simplifications to this, too :)